### PR TITLE
Small PRB props refactor

### PR DESCRIPTION
**Pull request by TomPretty**
## What are you doing in this PR?

Small refactor to the `StripePaymentRequestButtonContainer`s props. Currently it takes the `selectedAmounts` and `otherAmounts` and computes the `amount`. All it actually needs to know is the `amount`. Really `selectedAmounts` & `otherAmounts` should really just be the concern of the form components that render frequency & amount choice cards. I've refactored the ``StripePaymentRequestButtonContainer` to accept just the `amount` as a prop which makes it easier to share going forward (i.e in the new in epic checkout form).

### CODE REVIEW

File: support-frontend/assets/components/StripePaymentRequestButton/StripePaymentRequestButtonContainer.tsx [hunk 1]
```diff
@@ -2,32 +2,24 @@
...
 import { isInStripePaymentRequestAllowedCountries } from 'helpers/internationalisation/country';
-import type { IsoCurrency } from 'helpers/internationalisation/currency';
 import type { RenderPaymentRequestButton } from './StripePaymentRequestButton';
 import StripePaymentRequestButton from './StripePaymentRequestButton';
 
 // ----- Types -----//
 
 type PropTypes = {
 	country: IsoCountry;
-	currency: IsoCurrency;
```
Comment (TomPretty) [inline support-frontend/assets/components/StripePaymentRequestButton/StripePaymentRequestButtonContainer.tsx:26]: This props was unused so deleted it during this refactor too

File: support-frontend/assets/components/StripePaymentRequestButton/StripePaymentRequestButtonContainer.tsx [hunk 2]
```diff
@@ -38,32 +30,26 @@ interface PrbObjects {
...
 
-function StripePaymentRequestButtonContainer(
-	props: PropTypes,
-): JSX.Element | null {
+function StripePaymentRequestButtonContainer({
+	country,
+	contributionType,
+	isTestUser,
+	amount,
+	renderPaymentRequestButton,
```
Comment (TomPretty) [inline support-frontend/assets/components/StripePaymentRequestButton/StripePaymentRequestButtonContainer.tsx:38]: Thought I'd destructure the props whilst I was at it to be more inline with how we write new components

File: support-frontend/assets/pages/contributions-landing/components/StripePaymentRequestButton.tsx [hunk 3]
```diff
@@ -6,40 +6,29 @@ import React from 'react';
...
 }: StripePaymentRequestButtonProps): JSX.Element {
 	return (
 		<StripePaymentRequestButtonContainer
 			country={country}
-			currency={currency}
 			isTestUser={isTestUser}
 			contributionType={contributionType}
-			selectedAmounts={selectedAmounts}
-			otherAmounts={otherAmounts}
+			amount={amount}
```
Comment (TomPretty) [inline support-frontend/assets/pages/contributions-landing/components/StripePaymentRequestButton.tsx:31]: This component is specific to the contribs landing page (i.e won't be shared with the new in epic checkout) but I thought it made sense to also make it just recieve the `amount` as a prop as again that's all it needs to know

File: support-frontend/ts-error-history.csv [hunk 4]
```diff
@@ -59,4 +59,5 @@
 1638360727000,832,4a749cca7354efd2a096e7a7f3d53b3f43d1f951
 1638529562000,804,7cab8acd1b5c0b8aa6ba5bc50558cc2c476e19c1
 1638534645000,777,dcadc16cb35c15aba0bcd17b91133138892daf6f
-1638547419000,777,HEAD
+1638788405000,777,35d02200ed882b81b67a788f939b912ecb45e516
+1638791462000,777,HEAD
```
Comment (TomPretty) [inline support-frontend/ts-error-history.csv:63]: I wonder if it's worth while updating the script to only add new datapoints if it's different to the previous one?

**Review by GHaberis — APPROVED** [review]
Looks good

**Pull request merged**